src/coreclr/debug/createdump/crashinfo.cpp

```diff
@@ -319,7 +319,16 @@ CrashInfo::InitializeDAC(DumpType dumpType)
     m_dacModule = dlopen(dacPath.c_str(), RTLD_LAZY);
     if (m_dacModule == nullptr)
     {
-        printf_error("InitializeDAC: dlopen(%s) FAILED %s\n", dacPath.c_str(), dlerror());
+        if (m_appModel == AppModelType::SingleFile)
+        {
+            printf_error("Only full dumps are supported by single file apps without the DAC module. Either copy %s to %s or change the dump type to full (DOTNET_DbgMiniDumpType=4)\n",
```

Comment:
Could we simplify this further and just say that officially single-file doesn't support minidumps (for now)? I worry that if we start telling people to do this we'll get:
1. Complaints that our design defeats the whole purpose of single-file if they have to copy multiple files.
2. Questions/issues from people who don't know where to copy mscordaccore from or they copy the wrong version of the file and it still doesn't work.

Reply:
You are right this error message should be just about only full dumps supported (I'll change the message).  I think having this (advanced) option to copy the DAC to get heap dumps is important and I don't want to remove this feature just in case we have a customer that really needs heap dumps. Hopefully you are ok with leaving it an undocumented feature.